checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleConfiguration.java

```diff
@@ -86,7 +86,11 @@ public File getXmlDefinitionFile() {
     Iterable<File> files = fileSystem.files(predicates.and(
       predicates.hasLanguage(CheckstyleConstants.JAVA_KEY),
       predicates.hasType(InputFile.Type.MAIN)));
-    return ImmutableList.<File>builder().addAll(files).build();
+    List<File> fileList = new ArrayList<>();
+    for (File file : files) {
+      fileList.add(file);
+    }
+    return fileList;
```

Comment:
The ownership of the list transfers from the callee to the caller. Modifying the list does not modify or even corrupt the state of the callee. The same is true for the list below.